tracking/inc/track.h

```diff
@@ -15,28 +17,153 @@ enum TrackStatus { alive, dead };
 
 struct track {
   RanluxppDouble rng_state;
-  int index{0};
-  int pdg{0};
-  double energy{10};
+  unsigned int   index{0};
+  double         energy{10};
+   
   vecgeom::Vector3D<double> pos;
   vecgeom::Vector3D<double> dir;
-  vecgeom::NavStateIndex current_state;
-  vecgeom::NavStateIndex next_state;
-  int mother_index{0};
-  TrackStatus status{alive};
-  int current_process{0};
+  vecgeom::NavStateIndex    current_state;
+  vecgeom::NavStateIndex    next_state;
+
+  unsigned int mother_index{0};
+  unsigned int eventId{0};
+  TrackStatus  status{alive};
+
   float interaction_length{FLT_MAX};
-  float energy_loss{0};         // primitive version of scoring
-  int number_of_secondaries{0}; // primitive version of scoring
+  float total_length{0.0};        // length since track start
+   
+  float energy_loss{0};           // primitive version of scoring
+  int   number_of_secondaries{0}; // primitive version of scoring
 
-  __device__ double uniform() { return rng_state.Rndm(); }
+  uint16_t    num_step{0}; // number of Steps -- for killing after MaX
+   
+  char        current_process{0}; // -1 for Navigation / -2 for Field Propagation
+  char        pdg{0};             // Large enough for e/g. Then uint16_t for p/n
 
-  __device__ void SwapStates()
+  __device__ __host__ double uniform() { return rng_state.Rndm(); }
+
+  __device__ __host__ int    charge() const  // charge for e- / e+ / gamma only
+  { constexpr char pdgElec= 11;
+    int chrg= (pdg== -pdgElec ? 1 : 0)+(pdg==  pdgElec ? -1 : 0 ); return chrg; }
```

Comment:
Can you please run `clang-format` on this PR, but at least on the modified files?

Reply:
I have a difficulty with naming this 'number_of_secondaries' if it is really something else.

Can we either a) adapt the scoring code to the real use case (seems relatively easy - it's one line?) or b) rename thje one for scoring and keep the name 'number_of_secondaries' for its correct meaning ?
For example, the RNG should skip by number_of_secondaries * skip_value ( = 1 << 15 now) , not a constant 1<<15 which would make sequences for subsequent tracks overlap with each other.

If it's supposed to flag whether the track has created more than zero secondaries, the current 'naive' scoring can simply check whether the 'true' number_of_secondaries is above zero

Reply:
It seems that the current use was more 'produced_seconaries_in_current_step' -- but as far as I can tell it was never cleared.
So a track that had ever produced a secondary in a previous step, was thought to have produced a secondary in the current step ...

Reply:
I have revised the 'scoring' in example2.cu to try to restore the previous result - simply 'squashing' the value of number_of_secondaries: if it is > 0 it is counted as 1.
Does that restore the counting that you expected (for now?)

Reply:
see my reply above:

> Unfortunately no, because that would assume a process creates secondaries in every step after it has created at least one...

Reply:
I revised the code in example2 to fetch the previous number of secondaries, before an interaction, and the current after it.  It will add the number of secondaries (only if it is positive - to avoid adding zero. )  
Changed energy_loss.h (no longer zeroes number_of_secondaries) and pair_production.h (uses ++).
This now reproduces the old output.